caravel/sql_lab.py

```diff
@@ -135,23 +119,32 @@ def handle_error(msg):
         query.select_sql = '{}'.format(database.select_star(
             query.tmp_table_name, limit=query.limit))
     query.end_time = utils.now_as_float()
+    session.flush()
+
+    payload = {
+        'query_id': query.id,
+        'status': query.status,
+        'data': [],
+    }
+    if query.status == models.QueryStatus.SUCCESS:
+        payload['data'] = cdf.data if cdf else []
+        payload['columns'] = cdf.columns_dict if cdf else []
+        query.state = 'success'
```

Comment:
I just realized that there's no code path for failed queries here, if the query fails the code never makes it here... I removed a bunch of lines